src/lib/core/datetime.c

```diff
@@ -289,3 +289,91 @@ interval_to_string(const struct datetime_interval *ival, char *buf, ssize_t len)
 	}
 	return sz;
 }
+
+#define CHECK_RANGE(v, from, to) \
+	if ((v) < (from))	return -1; \
+	else if ((v) > (to))	return +1;
+
+#define CHECK_DT(dt) \
+	CHECK_RANGE(dt, INT_MIN, INT_MAX)
```

Comment:
I think you can drop the first macro and convert the second one to a static inline function, but that's up to you.

Reply:
Nope, in this case `return` would stop to work